### ISSUE TITLE
Current file has newer changes on other branches

### ISSUE THREAD

**Issue by devinrhode2** [eamodio/vscode-gitlens#1422]
Idea is simple, but not trivial to implement. For any given file I have open, I want a little orange indicator in the status bar if the file has newer changes (based on git commit timestamp) on other branches.

If teams commit+push early and often, then this can help you predict (and therefore, minimize/prevent) merge conflicts :)

**Comment by devinrhode2** [comment]
An extra level could be - if the current (saved|unsaved file) _actually would_ have merge conflicts with other branches, then the indicator would be red (or, a more alarming color of your choosing). Might turn into a whole "merge conflict predictor/preview" sidebar, with it's own dedicated UI...

But I think the most useful thing is trying to _prevent_ merge conflicts from happening in the first place, by helping developers see newer changes from other branches before doing too much work on any given file. Once everyone on a team is using this little feature to avoid creating merge conflicts in the first place, there's way less motivation for managing/handling/helping with large merge conflicts.

**Comment by devinrhode2** [comment]
Funny: the way I would use this right now, if the feature existed, would be to "publish" a nice refactor to a file, which doesn't really accomplish any end goals, and is still a work in progress, which maybe I shouldn't have been working on in the first place, but the code seemed a little whack - so I just had to shuffle things around until I mapped everything out clearly, and then kept on cleaning things up to make it easier for me to read. It's a horrible use case, and am only mentioning it for comedic purposes. Anyone could quickly make a small change to the file in the future and then my change would be "older" than the default branch, and probably end up lost and forgotten. Interesting to ponder tho. I should probably get the refactor merged, even if it's not really valuable to end-users.

**Comment by eamodio** [comment]
Good idea -- one I've thought about before and would love to spend more time figuring out the best way to provide this.

**Comment by devinrhode2** [comment]
> <sub>The way I would use this right now, if the feature existed, would be to "publish" a nice refactor to a file, which doesn't really do anything for end users, and is still a work in progress, which maybe I shouldn't have been working on in the first place, but the code seemed a little whack - I should probably get the refactor merged, even if it's not really valuable to end-users.</sub>

Actually, refactors like this could be published with the branch prefix `pending-refactors/`. Essentially, they would be somewhat hidden draft PRs. Since, they have no effect on the business today, there's no point in reviewing them. However, next time someone goes to edit that code, they'd get a red or green dot, indicating there's been changes to this file.

It's tough, because really refactors should just get reviewed and merged, so code is more readable. But it may just not be the highest priority thing for a team/business.
